src/main/java/com/cleanroommc/groovyscript/compat/mods/lightningcraft/Infusion.java

```diff
@@ -0,0 +1,133 @@
+    })
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Property(property = "input", comp = @Comp(gte = 0, lte = 4))
+    @Property(property = "output", comp = @Comp(eq = 1))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<LightningInfusionRecipe> {
+
+        @Property()
+        private IIngredient centerItem = null;
+
+        @Property(comp = @Comp(gte = 0), defaultValue = "-1")
+        private int le = -1;
+
+        @Property(defaultValue = "determined automatically based on the input items")
+        private boolean nbtSensitive = false;
+
+        private boolean nbtSensitiveChanged = false;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder le(int le) {
+            this.le = le;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription(field = "le")
+        public RecipeBuilder cost(int le) {
+            this.le = le;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder nbtSensitive(boolean nbtSensitive) {
+            this.nbtSensitive = nbtSensitive;
+            nbtSensitiveChanged = true;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder centerItem(IIngredient centerItem) {
+            this.centerItem = centerItem;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding LightningCraft Infusion Table recipe";
+        }
+
+        @Override
+        protected int getMaxItemInput() {
+            // recipes with more than 1 item in some slot don't get recognized
+            return 1;
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 0, 4, 1, 1);
+            validateFluids(msg);
+            msg.add(centerItem == null, "Center item must not be null");
+            msg.add(centerItem != null && centerItem.getMatchingStacks().length == 0, "Center item must not have a matching item");
+            msg.add(le < 0, "LE cost must be positive");
+            for (IIngredient it : this.input) {
+                msg.add(it == null || it.getMatchingStacks().length == 0, "All inputs must have a matching item");
+            }
+            if (GroovyScriptConfig.compat.checkInputStackCounts && centerItem != null) {
+                msg.add(centerItem.getAmount() > 1, "Expected stack size of 1 for {}, got {}", centerItem, centerItem.getAmount());
+            }
+        }
+
+        @Override
+        @RecipeBuilderRegistrationMethod
+        public @Nullable LightningInfusionRecipe register() {
+            if (!validate()) return null;
+
+            ItemStack centerItem = this.centerItem.getMatchingStacks()[0];
+            ItemStack[] inputs = input.stream().map(i -> i.getMatchingStacks()[0]).toArray(ItemStack[]::new);

```

Comment:
while it is not currently on `main` and is still just in the Betweenlands PR, `IngredientHelper.cartesianProductItemStacks` would be very useful here to, once again, use all values of `getMatchingStacks()` instead of just the first.